### PR TITLE
DMP-1746-Urgency-Sort

### PR DESCRIPTION
**Pull request by bpires2**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/DMP-1746

### Change description ###

- Data table can now be sorted by a custom sort function
- Urgencies column can be sorted in terms of priority

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/app/types/transcription-urgency.interface.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export interface TranscriptionUrgency {
   transcription_urgency_id: number;
   description: string;
+  priority_order?: number;
```
Comment (davet1985) [inline src/app/types/transcription-urgency.interface.ts:4]: Not sure this is optional?

File: src/app/services/custom-sort/table-custom-sort-functions.service.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import { Order } from '@darts-types/data-table-column.interface';
+import { TranscriptionUrgency } from '@darts-types/transcription-urgency.interface';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TableCustomSortFunctionsService {
+  // only works for transcription data table
+  sortByUrgencyPriorityOrder(a: unknown, b: unknown, direction?: Order) {
```
Comment (davet1985) [inline src/app/services/custom-sort/table-custom-sort-functions.service.ts:10]: Can you not specify the types in the parameters?

```suggestion
  sortByUrgencyPriorityOrder(a: { urgency: TranscriptionUrgency }, b: { urgency: TranscriptionUrgency }, direction?: Order) {
```

File: src/app/types/transcription-urgency.interface.ts (hunk 1 above)
Comment (bpires2) [reply src/app/types/transcription-urgency.interface.ts:4]: @deanbullock-hmcts suggested that it be optional

File: src/app/services/custom-sort/table-custom-sort-functions.service.ts (hunk 2 above)
Comment (bpires2) [reply src/app/services/custom-sort/table-custom-sort-functions.service.ts:10]: if we specify in the parameters it breaks the constraints of the interface we have in place so we have to do it this way sadly and type cast afterwards inside the function

File: src/app/types/transcription-urgency.interface.ts (hunk 1 above)
Comment (davet1985) [reply src/app/types/transcription-urgency.interface.ts:4]: Argh, it isn't marked as required in the openapi spec, so I guess this is fine.

**Review by davet1985 — APPROVED** [review]

**Pull request merged**